packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts

```diff
@@ -75,38 +85,93 @@ const calculate = (
     return payloadData;
 };
 
+const fetchAccountOwnerAndTokenInfoForAddress = async (
+    address: string,
+    symbol: string,
+    mint: string,
+) => {
+    // Fetch data about recipient account owner if this is a token transfer
+    // We need this in order to validate the address and ensure transfers go through
+    let accountOwner: string | undefined;
+    let tokenInfo: TokenAccount[] | undefined;
+
+    const accountInfoResponse = await TrezorConnect.getAccountInfo({
+        coin: symbol,
+        descriptor: address,
+    });
+
+    if (accountInfoResponse.success) {
+        accountOwner = accountInfoResponse.payload?.misc?.owner;
+        tokenInfo = accountInfoResponse.payload?.tokens?.find(
+            token => token.contract === mint,
+        )?.accounts;
+    }
+
+    return [accountOwner, tokenInfo] as const;
+};
+
 export const composeTransaction =
     (formValues: FormState, formState: ComposeActionContext) =>
     async (_dispatch: Dispatch, getState: GetState) => {
         const { account, network, feeInfo } = formState;
         const composeOutputs = getExternalComposeOutput(formValues, account, network);
         if (!composeOutputs) return; // no valid Output
 
-        const { output, decimals } = composeOutputs;
+        const { output, decimals, tokenInfo } = composeOutputs;
 
         let fetchedFee: string | undefined;
 
         const blockhash = getState().wallet.blockchain.sol.blockHash;
 
+        const [recipientAccountOwner, recipientTokenInfo] = tokenInfo
+            ? await fetchAccountOwnerAndTokenInfoForAddress(
+                  formValues.outputs[0].address,
+                  account.symbol,
+                  tokenInfo.contract,
+              )
+            : [undefined, undefined];
```

Comment:
This `[undefined, undefined]` seems unnecessary and not very nice,  so I would remove `fetchAccountOwnerAndTokenInfoForAddress` and just do 

```
    const accountInfoResponse = await TrezorConnect.getAccountInfo({
        coin: symbol,
        descriptor: address,
    });

    const accountOwner = accountInfoResponse.payload?.misc?.owner;
    const tokenInfo = accountInfoResponse.payload?.tokens?.find(
            token => token.contract === mint,
        )?.accounts;
```

Also btw isn't accountInfo already in `state` ? Do we need to refetch it here?

Reply:
this is for the *recipient*, this account comes from the form field. The `[undefined, undefined]` part is a fallback so that we don't fetch when we're not sending tokens == don't need this data.